examples/fabric/kfold_cv/train_fabric.py

```diff
@@ -0,0 +1,199 @@
+# Copyright The Lightning AI team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+from os import path
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import torch.optim as optim
+import torchvision.transforms as T
+from sklearn import model_selection
+from torch.utils.data import DataLoader, SubsetRandomSampler
+from torchmetrics.classification import Accuracy
+from torchvision.datasets import MNIST
+
+from lightning.fabric import Fabric  # import Fabric
+from lightning.fabric import seed_everything
+
+DATASETS_PATH = path.join(path.dirname(__file__), "..", "..", "..", "Datasets")
+
+
+class Net(nn.Module):
+    def __init__(self) -> None:
+        super().__init__()
+        self.conv1 = nn.Conv2d(1, 32, 3, 1)
+        self.conv2 = nn.Conv2d(32, 64, 3, 1)
+        self.dropout1 = nn.Dropout(0.25)
+        self.dropout2 = nn.Dropout(0.5)
+        self.fc1 = nn.Linear(9216, 128)
+        self.fc2 = nn.Linear(128, 10)
+
+    def forward(self, x):
+        x = self.conv1(x)
+        x = F.relu(x)
+        x = self.conv2(x)
+        x = F.relu(x)
+        x = F.max_pool2d(x, 2)
+        x = self.dropout1(x)
+        x = torch.flatten(x, 1)
+        x = self.fc1(x)
+        x = F.relu(x)
+        x = self.dropout2(x)
+        x = self.fc2(x)
+        output = F.log_softmax(x, dim=1)
+        return output
+
+
+def train_dataloader(model, data_loader, optimizer, fabric, epoch, hparams, fold):
+    # TRAINING LOOP
```

Comment:
@shenoynikhil Very naive question (because [I was interested in this feature at some point](https://github.com/Lightning-AI/lightning/issues/839#issuecomment-808489228)): one of the key arguments in favor of lightning is that it removes boilerplate code; then how come you're writing you own training/evaluation loops and not using `pl.Trainer`?

Reply:
When I read this, I feel like there's really no advantage to using lightning over [accelerate](https://github.com/huggingface/accelerate)! (but there should be, right?) :-)

Reply:
So there are two ways to include lightning in your code. First, is using the PL framework end-to-end (Training, LightningModule, etc), this removes most of the boilerplate code as you said and allows you to do advanced stuff like distributed training, etc. Second, is using lightning [fabric](https://pytorch-lightning.readthedocs.io/en/stable/fabric/fabric.html) which allows you to modify your existing pytorch code to enable it many of the advantages provided by PL (read the documentation for more info on how this is different from the first part).

Now coming to KFold cross validation, implementing this the first way is a bit challenging since it requires changes to the `pl.Trainer`. So an easier way to still power KFold cross validation with pl would be to use fabric. @awaelchli might be able to add more on this :). Hope I have still answered some part of your question.

Reply:
Understood, thanks! I believe that the real added value would be to make `pl.Trainer` work with CV (or nested CV). As you noted, that's the complicated part...